packages/components/src/components/tab-nav/tab-nav.tsx

```diff
@@ -42,9 +42,16 @@ export class TabNav {
   handleSelect(event) {
     const nextTab = event.target as HTMLScaleTabHeaderElement;
     // Act only if it's a direct child
-    if (this.getAllEnabledTabs().includes(nextTab) && !nextTab.disabled) {
-      this.selectTab(nextTab);
+    if (!this.getAllEnabledTabs().includes(nextTab) || nextTab.disabled) {
```

Comment:
This way, the added return call will be useless. I suggest you return this to its previous form and adapt the `selectTab`